### PR TITLE
Cleanup and deprecates autoindent flags and options.

### PR DESCRIPTION
**Pull request by Carreau**
cc @takluyver

### CODE REVIEW

File: IPython/terminal/magics.py [hunk 1]
```diff
@@ -84,8 +84,7 @@ def autoindent(self, parameter_s = ''):
         """Toggle autoindent on/off (deprecated)"""
         print("%autoindent is deprecated since IPython 5: you can now paste "
               "multiple lines without turning autoindentation off.")
-        self.shell.set_autoindent()
-        print("Automatic indentation is:",['OFF','ON'][self.shell.autoindent])
+        print("Automatic indentation is deprecated and will have no effects.")
```
Comment (takluyver) [inline IPython/terminal/magics.py]: I don't think we need two messages about the same deprecation.

**Comment by takluyver** [comment]
I'm having trouble seeing tests on Travis at the moment, but the test failures on Appveyor look real.

**Comment by Carreau** [comment]
@ivanov I'm going to guess we should not remove that if you'r going to bring back readline ?

**Comment by unode** [comment]
I was looking for a way to disable autoindentation and found `--no-autoindent` but this option has no effect (IPython 6.1.0 & 6.2.1).
Should the deprecation in this issue also affect `--no-autoindent` and `--autoindent`? Those do not produce any warning.

**Pull request closed without merging**